### ISSUE TITLE
apigatewaymanagementapi postToConnection failing

### ISSUE THREAD

**Issue by chirag729** [agilord/aws_client#312]
postToConnection is failing with 403 InvalidSignatureException.

I managed to resolve the issue with the following modification to line 109 of apigatewaymanagementapi-2018-11-29.dart

`requestUri: '/@connections/${Uri.encodeComponent(connectionId)}',`

`requestUri: '/${Uri.encodeComponent('@connections')}/${Uri.encodeComponent(connectionId)}',`

**Comment by Schwusch** [comment]
I found a [TODO I wrote 2 years and 3 days ago](https://github.com/agilord/aws_client/blob/master/generator/lib/builders/protocols/service_builder.dart#L54):
```dart
// TODO implement UriEscape and UriEscapePath according to https://github.com/aws/aws-sdk-js/blob/master/lib/util.js#L39..L57
```
Which points to this snippet:
```js
uriEscape: function uriEscape(string) {
    var output = encodeURIComponent(string);
    output = output.replace(/[^A-Za-z0-9_.~\-%]+/g, escape);

    // AWS percent-encodes some extra non-standard characters in a URI
    output = output.replace(/[*]/g, function(ch) {
      return '%' + ch.charCodeAt(0).toString(16).toUpperCase();
    });

    return output;
  },

  uriEscapePath: function uriEscapePath(string) {
    var parts = [];
    util.arrayEach(string.split('/'), function (part) {
      parts.push(util.uriEscape(part));
    });
    return parts.join('/');
  },
```
Seems like it's time to actually implement that one :sweat_smile:

**Comment by Schwusch** [comment]
This should be resolved in aws_apigatewaymanagementapi_api 1.0.1
